repos/keg-components/src/components/list/sectionList.native.js

```diff
@@ -0,0 +1,394 @@
+ * @param {Array} props.sections - Groups of items to be displayed in the SectionList component
+ * @param {Object} props.styles - Merged custom and theme styles
+ */
+const SectionHeader = props => {
+  const {
+    index,
+    onSectionChange,
+    renderSectionHeader,
+    sectionRefs,
+    section,
+    styles,
+  } = props
+
+  const sectionRef = useRef(null)
+
+  useEffect(() => {
+    // If no onSectionChange method
+    // then we don't need to set the sectionRefs
+    if (!isFunc(onSectionChange)) return
+
+    sectionRefs.current[index] = sectionRef.current
+
+    sectionRefs.current[index].element &&
+      !sectionRefs.current[index].layout &&
+      (sectionRefs.current[index].layout = getElementLayout(
+        sectionRef.current.element
+      ))
+
+    return () => delete sectionRefs.current[index]
+  }, [ sectionRef.current, index, onSectionChange ])
+
+  // Wrap the renderSectionHeader method
+  // With a fragment, and view so we can track the section headers
+  // with an internal component and ref
+  return (
+    <>
+      <View
+        className={`keg-section-${index}`}
+        ref={element => (sectionRef.current = { element, index })}
+      />
+      { checkCall(renderSectionHeader, { section, styles, onSectionChange }) }
+    </>
+  )
+}
+
+/**
+ * Helper hook for the renderSectionHeader, that wraps our custom Section header in a callback
+ * @param {function} renderSectionHeader - Consumers callback method to render their section headers
+ * @param {function} onSectionChange - Consumers callback called whe a section is changed
+ * @param {Object} sectionRefs - React ref of all Section divider components.
+ * @param {Object} styles - Merged custom and theme styles
+ *
+ * @returns {function} - Render function for Section Headers
+ */
+const useRenderSectionHeader = (
+  renderSectionHeader,
+  onSectionChange,
+  sectionRefs,
+  styles
+) => {
+  // This gets used by the renderSectionHeader prop
+  // It's not an issue with hooks, because the Callback hook
+  // Has already been called before is passed as prop to the SectionList component
+  // Which means we can call hooks in sub components
+  return useCallback(
+    ({ section }) => {
+      return (
+        <SectionHeader
+          index={section.__kegIndex}
+          renderSectionHeader={renderSectionHeader}
+          onSectionChange={onSectionChange}
+          section={section}
+          sectionRefs={sectionRefs}
+          styles={styles}
+        />
+      )
+    },
+    [ styles, onSectionChange, sectionRefs.current, renderSectionHeader ]
+  )
+}
+
+const useRenderItem = (renderItem, onSectionChange) => {
+  return useCallback(
+    ({ item }) => {
+      return checkCall(renderItem, { item, onSectionChange })
+    },
+    [ renderItem, onSectionChange ]
+  )
+}
+
+/**
+ * SectionList
+ * @summary Default view component that wraps the React Native View component. All props are optional
+ *
```

Comment:
i think it could be helpful to at least add jsdoc description for the callbacks `onScrollSectionChange` and `onSectionChange` for the props that will be passed to the consumer's callback func.

if we're going with the approach of `see x PropTypes`
apparently we can just add the description there and itll show up in the storybook:
something like this:
- [image](https://user-images.githubusercontent.com/3317835/104960563-b7941f80-5991-11eb-9abf-4e0fbe3e6a6f.png)
- [image](https://user-images.githubusercontent.com/3317835/104960611-cda1e000-5991-11eb-99dd-d4a46fe610b1.png)

Reply:
Updated